ui/shared/ContactSelector.qml

```diff
@@ -0,0 +1,187 @@
+import QtQuick 2.13
+import QtQuick.Controls 2.13
+import QtQuick.Layouts 1.13
+import QtGraphicalEffects 1.13
+import "../imports"
+
+Item {
+    id: root
+    property var contacts
+    property var selectedContact
+    height: select.height + (validationErrorText.visible ? validationErrorText.height : 0)
+    property int dropdownWidth: width
+    property string validationError: validationErrorText.text
+    property alias validationErrorAlignment: validationErrorText.horizontalAlignment
+
+    onContactsChanged: {
+        root.selectedContact = { name: qsTr("Select a contact") }
+    }
+
+    function validate() {
+        const isValid = root.selectedContact && root.selectedContact.address
+        if (!isValid) {
+            select.select.border.color = Style.current.danger
+            select.select.border.width = 1
+            validationErrorText.visible = true
+        } else {
+            select.select.border.color = Style.current.transparent
+            select.select.border.width = 0
+            validationErrorText.visible = false
```

Comment:
Note, for the Input, I created a validationError prop. Maybe it's worth adding it to the Select as well so that it handles its borders itself.

Reply:
Good idea, updated the Select component to handle the validation!